cmd/scollector/collectors/google_analytics.go

```diff
@@ -0,0 +1,99 @@
+package collectors
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/gob"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	analytics "bosun.org/_third_party/google.golang.org/api/analytics/v3"
+
+	"bosun.org/_third_party/golang.org/x/net/context"
+	"bosun.org/_third_party/golang.org/x/oauth2"
+	"bosun.org/_third_party/golang.org/x/oauth2/google"
+	"bosun.org/cmd/scollector/conf"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+func init() {
+	registerInit(func(c *conf.Conf) {
+		for _, g := range c.GoogleAnalytics {
+			collectors = append(collectors, &IntervalCollector{
+				F: func() (opentsdb.MultiDataPoint, error) {
+					return c_google_analytics(g.ClientID, g.Secret, g.Token, g.Sites)
+				},
+				name: "c_google_analytics",
+			})
+		}
+	})
+}
+
+func c_google_analytics(clientid string, secret string, tokenstr string, sites []conf.GoogleAnalyticsSite) (opentsdb.MultiDataPoint, error) {
+	var md opentsdb.MultiDataPoint
+
+	c, err := analyticsClient(clientid, secret, tokenstr)
+	if err != nil {
+		return nil, err
+	}
+	svc, err := analytics.New(c)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, site := range sites {
+		call := svc.Data.Realtime.Get("ga:"+site.Profile, "rt:pageviews").Dimensions("rt:minutesAgo")
+		data, err := call.Do()
+		if err != nil {
+			return md, err
+		}
+		time := time.Now().Add(time.Duration(-1*site.Offset) * time.Minute).Unix()
+		value := "0"
+		// Iterates through the response data and returns the time slice we
+		// actually care about when we find it. If no offset was specified,
+		// the row we care about is minute '0' - or the most recently
+		// gathered datapoint.
+		for _, v := range data.Rows {
+			minute, err := strconv.Atoi(v[0])
+			if err != nil {
+				return md, fmt.Errorf("Error parsing GA data: %s", err)
+			}
+			if minute == site.Offset {
+				value = v[1]
+				break
+			}
+		}
+		AddTS(&md, "google.analytics.realtime.pageviews", time, value, opentsdb.TagSet{"site": site.Name}, metadata.Gauge, metadata.Count, "Number of pageviews tracked by GA in one minute")
```

Comment:
Yeah, it's a string representation of a number. I can add a type assertion and convert it.